lms/djangoapps/courseware/views.py

```diff
@@ -1419,3 +1426,180 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
             'xqa_server': settings.FEATURES.get('XQA_SERVER', 'http://your_xqa_server.com'),
         }
         return render_to_response('courseware/courseware-chromeless.html', context)
+
+
+def get_analytics_answer_dist(request):
+    """
+    Calls the the analytics answer distribution api client. Retrieves answer distribution data for the in-line
+    analytics display.
+
+    Arguments:
+        request (django request object):  the HTTP request object that triggered this view function
+
+    Returns:
+        (django response object):  JSON response:
+            500 if error occurred,
+            404 if api client returns no data,
+            otherwise 200 with JSON body.
+    """
+
+    all_data = json.loads(request.POST['data'])
+    module_id = all_data['module_id']
+    question_types_by_part = all_data['question_types_by_part']
+    num_options_by_part = all_data['num_options_by_part']
+    course_key = SlashSeparatedCourseKey.from_string(all_data['course_id'])
+
+    # Construct an error message
+    if getattr(settings, 'ZENDESK_URL'):
+        link_start = "<a href=\"" + getattr(settings, 'ZENDESK_URL') + "/hc/en-us/requests/new\">"
+        link_text = _('here')
+        link_end = "</a>"
+        error_message = _("A problem has occurred retrieving the data, to report the problem click {link_start}{link_text}{link_end}").format(
+            link_start=link_start,
+            link_text=link_text,
+            link_end=link_end,
+        )
+    else:
+        error_message = _('A problem has occurred retrieving the data.')
+
+    # Check user is enrolled as a staff member of this course
+    try:
+        course = get_course_with_access(request.user, 'staff', course_key, depth=None)
+    except Http404:
+        return HttpResponseServerError(error_message)
+
+    having_access = has_access(request.user, 'staff', course)
+    url = getattr(settings, 'ANALYTICS_DATA_URL')
+    auth_token = getattr(settings, 'ANALYTICS_DATA_TOKEN')
+
+    if not having_access or not url:
+        return HttpResponseServerError(error_message)
+
+    client = Client(base_url=url, auth_token=auth_token)
+    module = client.modules(course.id, module_id)
+
+    try:
+        data = module.answer_distribution()
+    except NotFoundError:
+        return HttpResponseNotFound(_('There are no student answers for this problem yet; please try again later.'))
+    except InvalidRequestError:
+        return HttpResponseServerError(error_message)
+    except TimeoutError:
+        return HttpResponseServerError(error_message)
```

Comment:
Please make sure that the error message here is properly internationalized. See http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/internationalization/i18n.html#coverage-testing